assets/dev/js/editor/controls/media.js

```diff
@@ -222,27 +255,78 @@ ControlMediaItemView = ControlMultipleBaseItemView.extend( {
 		}
 
 		if ( attachment.url ) {
-			this.setValue( {
+			const updatedState = {
 				url: attachment.url,
 				id: attachment.id,
 				alt: attachment.alt,
 				source: attachment.source,
-			} );
+			};
+			this.setValue( updatedState );
 
 			if ( this.model.get( 'responsive' ) ) {
 				// Render is already calls `applySavedValue`, therefore there's no need for it in this case.
 				this.renderWithChildren();
 			} else {
 				this.applySavedValue();
 			}
-		}
 
+			this.setPrevState( updatedState );
+			this.toggleSizeControl();
+		}
 		this.trigger( 'after:select' );
 	},
 
 	onBeforeDestroy() {
 		this.$el.remove();
 	},
+
+	onMediaImageSizeChange() {
+		this.ui.mediaImageSize.removeClass( 'e-select-placeholder' );
+		const selectedImage = {
+			id: this.getCurrentValue()?.id ?? this.getEditSettings( 'mediaId' ),
+			alt: this.getCurrentValue()?.alt ?? this.getEditSettings( 'mediaAlt' ),
+		};
+
+		const selectedSize = this.ui.mediaImageSize.val();
+		let imageURL = elementor.imagesManager.getImageUrl( {
+			id: selectedImage.id,
+			size: selectedSize,
+		} );
+
+		const stateOptions = {
+			url: null,
+			id: selectedImage.id,
+			alt: selectedImage.alt,
+			dimensions: selectedSize,
+			source: 'library',
+		};
+
+		if ( imageURL ) {
+			stateOptions.url = imageURL;
+			this.setValue( stateOptions );
```

Comment:
You can exit from the function and remove the `else` statement.

Reply:
<img width="599" alt="Screen Shot 2022-12-06 at 13 44 17" src="https://user-images.githubusercontent.com/100698721/205902992-1acecd78-e9a8-4baa-88fc-da9b41270d15.png">

I also need the else to listen to `imagesManager:detailsReceived` hook, only on the first Ajax request to get the image data. You can also read my discussion with @StyleShit 

https://github.com/elementor/elementor/pull/20361#discussion_r1033167730

Reply:
Done